src/main/scala/io/github/kiemlicz/shelm/ChartDownloader.scala

```diff
@@ -33,18 +51,55 @@ object ChartDownloader {
           )
         else
           downloadDir / topDirs.head
-      case ChartLocation.AddedRepository(ChartName(name), ChartRepositoryName(repoName), chartVersion) =>
-        val options = s"$repoName/$name -d $downloadDir${chartVersion.map(v => s" --version $v").getOrElse("")} --untar"
-        IO.delete(downloadDir)
-        pullChart(options, sbtLogger)
-        downloadDir / name
-      case ChartLocation.RemoteRepository(ChartName(name), uri, settings, chartVersion) =>
-        val authOpts = HelmPlugin.chartRepositoryCommandFlags(settings)
-        val allOptions = s"--repo $uri $name $authOpts -d $downloadDir${chartVersion.map(v => s" --version $v").getOrElse("")} --untar"
-        IO.delete(downloadDir)
-        pullChart(allOptions, sbtLogger)
-        downloadDir / name
+      case ChartLocation.AddedRepository(chartName, ChartRepositoryName(repoName), chartVersion) =>
+        pullFromCacheOrRemote(
+          chartName, chartVersion, () => {
+            val options = s"$repoName/${chartName.name} -d $downloadDir${
+              chartVersion.map(v => s" --version $v").getOrElse("")
+            } --untar"
+            IO.delete(downloadDir)
+            pullChart(options, sbtLogger)
+          }
+        )
+      case ChartLocation.RemoteRepository(chartName, uri, settings, chartVersion) =>
+        pullFromCacheOrRemote(
+          chartName, chartVersion, () => {
+            val authOpts = HelmPlugin.chartRepositoryCommandFlags(settings)
+            val allOptions = s"--repo $uri ${chartName.name} $authOpts -d $downloadDir${
+              chartVersion.map(v => s" --version $v").getOrElse("")
+            } --untar"
+            IO.delete(downloadDir)
+            pullChart(allOptions, sbtLogger)
+          }
+        )
+    }
+  }
+
+  def calcCacheChartDir(
+    chartName: ChartName, chartVersion: Option[String], cacheBaseDir: File
+  ): File = chartVersion match {
+    case Some(version) => new File(cacheBaseDir, s"${chartName.name}-$version")
+    case None => new File(cacheBaseDir, s"${chartName.name}")
```

Comment:
`None` version means that we should pull the latest version, right?
In such scenario the cache breaks the `always-latest-version` semantics, and we shouldn't use cache fin such situation IMO